components/button/button-mixin.js

```diff
@@ -108,7 +108,7 @@ export const ButtonMixin = superclass => class extends FocusVisiblePolyfillMixin
 	}
 
 	focus() {
-		const button = this.shadowRoot.querySelector('button');
+		const button = this.shadowRoot ? this.shadowRoot.querySelector('button') : undefined;
```

Comment:
If Dave L can get optional chaining to be allowed in our Polymer builds, it will be very easy to move over to the ?. chaining syntax. Until then we are stuck with these ugly checks 😢 . I don't think we will have these in place for most of the new code written, as we can better audit to know when exactly the code is called.

Reply:
```suggestion
		const button = this.shadowRoot && this.shadowRoot.querySelector('button');
```
Any chance you tried this? Its a bit smaller and I _think_ it should work the same

Reply:
Sorry wasn't implying for you to go change all these, just suggesting a potentially alternate for the future. I try and avoid `?:` at all costs

Reply:
Good catch, I will try that! Even if it's just a bit shorter it will help clean things up